airbyte-cdk/python/airbyte_cdk/sources/declarative/declarative_stream.py

```diff
@@ -114,7 +114,7 @@ def get_json_schema(self) -> Mapping[str, Any]:  # type: ignore
 
     def stream_slices(
         self, *, sync_mode: SyncMode, cursor_field: Optional[List[str]] = None, stream_state: Optional[Mapping[str, Any]] = None
-    ) -> Iterable[Optional[Mapping[str, Any]]]:
+    ) -> Iterable[Optional[PerPartitionStreamSlice]]:
```

Comment:
This seems like a breaking change if we assume CDK code will use the new interface. Do we have a plan to upgrade that?

Reply:
breaking change for what? existing consumers expect a `Mapping[str, Any]` which `PerPartitionStreamSlice` respects

Reply:
Ok, then the only problem I have is that parent classes will always have to be declarative for a declarative source which seems a bit restrictive given that we wanted to do partial migration to low-code. I guess this is the cost of only changing this to the declarative package and not an interface problem (basically, the discrepancy between the interface from this and `Stream` is just a symptom of this problem)